### PR TITLE
ci: [CNI] [NPM] Add NPM|CNI integration test to load-test pipeline.

### PR DESCRIPTION
**Pull request by jpayne3506**
**Reason for Change**:

Adds NPM|CNI test coverage to load-test pipeline.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] relevant PR labels added

**Notes**:

### CODE REVIEW

**Review by huntergregory — COMMENTED** [review]
lgtm

File: .pipelines/cni/k8s-e2e/k8s-e2e-job-template.yaml [hunk 1]
```diff
@@ -25,6 +25,7 @@ jobs:
             # sig-release provides test suite tarball(s) per k8s release. Just need to provide k8s version "v1.xx.xx"
             # pulling k8s version from AKS.
             eval k8sVersion="v"$( az aks show -g ${{ parameters.clusterName }}-$(make revision) -n ${{ parameters.clusterName }}-$(make revision) --query "currentKubernetesVersion")
+            echo $k8sVersion
```
Comment (vipul-21) [inline .pipelines/cni/k8s-e2e/k8s-e2e-job-template.yaml:28]: nit: Do we need this, we have the nextline as `https://dl.k8s.io/$k8sVersion)` which can tells us the used k8s version ?

File: .pipelines/cni/k8s-e2e/k8s-e2e-job-template.yaml [hunk 2]
```diff
@@ -50,7 +51,7 @@ jobs:
             name: dns
             clusterName: ${{ parameters.clusterName }}
             ginkgoFocus: '\[sig-network\].DNS.should'
-            ginkgoSkip: 'resolv'
+            ginkgoSkip: 'resolv|256 search'
```
Comment (vipul-21) [inline .pipelines/cni/k8s-e2e/k8s-e2e-job-template.yaml:54]: I think this based on a different PR?

File: .pipelines/cni/singletenancy/windows-cni-load-test-template.yaml [hunk 3]
```diff
@@ -33,7 +33,7 @@ stages:
     dependsOn: ${{ parameters.dependsOn }}
     displayName: "Build CNI Images"
     jobs:
-      - job: build_cni_images
+      - job: build_cni_images # Need to change name, will build various images from here (NPM|CNI)
```
Comment (vipul-21) [inline .pipelines/cni/singletenancy/windows-cni-load-test-template.yaml:36]: Let's do that then, change the job to `build_images`

File: .pipelines/cni/singletenancy/windows-cni-load-test-template.yaml [hunk 4]
```diff
@@ -43,17 +43,42 @@ stages:
...
               os_version: $(os_version)
+
+  - stage: binaries
+    displayName: Build Binaries
+    dependsOn: ${{ parameters.dependsOn }}
+    jobs:
+      - job:
+        displayName: "Build Kubernetes Test Suite Binary"
+        pool:
+          name: $(BUILD_POOL_NAME_DEFAULT) # May need to change
```
Comment (vipul-21) [inline .pipelines/cni/singletenancy/windows-cni-load-test-template.yaml:66]: `# May need to change` Why this comment ?

File: .pipelines/cni/k8s-e2e/k8s-e2e-job-template.yaml (hunk 2 above)
Comment (jpayne3506) [reply .pipelines/cni/k8s-e2e/k8s-e2e-job-template.yaml:54]: It is based off #2014, but I split it off as it affects multiple pipelines. Did not want it to get stuck within Load Test where it was found.

File: .pipelines/cni/k8s-e2e/k8s-e2e-job-template.yaml (hunk 1 above)
Comment (jpayne3506) [reply .pipelines/cni/k8s-e2e/k8s-e2e-job-template.yaml:28]: This step does not show the full curl command, only the results. The other way we could do this is change line 22 to` set -ex`.

File: .pipelines/cni/singletenancy/windows-cni-load-test-template.yaml (hunk 4 above)
Comment (jpayne3506) [reply .pipelines/cni/singletenancy/windows-cni-load-test-template.yaml:66]: Test comments that I missed. Wanted to confirm that `$(BUILD_POOL_NAME_DEFAULT)` matched here and also in PR pipeline for the variable group.

File: .pipelines/cni/singletenancy/windows-cni-load-test-template.yaml [hunk 5]
```diff
@@ -137,12 +162,30 @@ stages:
...
+    jobs:
+      - template: ../../npm/npm-cni-integration-test.yaml
+        parameters:
+          clusterName: ${{ parameters.clusterName }}
+          os: ${{ parameters.os }}
+          sub: $(TEST_SUB_SERVICE_CONNECTION)
+          os_version: 'ltsc2022'
+          tag: $(npmVersion)
+
+
```
Comment (vipul-21) [inline .pipelines/cni/singletenancy/windows-cni-load-test-template.yaml:182]: nit: extra line

File: .pipelines/npm/npm-cni-integration-test.yaml [hunk 6]
```diff
@@ -3,8 +3,10 @@ parameters:
   os: ""
   sub: ""
   os_version: ""
+  dependsOn: ""
   tag: ""
 
+
```
Comment (vipul-21) [inline .pipelines/npm/npm-cni-integration-test.yaml:9]: nit: whitespace

**Review by vipul-21 — APPROVED** [review]

**Review by vipul-21 — APPROVED** [review]

**Review by huntergregory — APPROVED** [review]

**Pull request merged**